src/main/kotlin/io/github/mojira/arisa/modules/TransferVersionsModule.kt

```diff
@@ -14,20 +14,28 @@ class TransferVersionsModule : AbstractTransferFieldModule<List<Version>, String
         field: List<Version>
     ) =
         parents.flatMap { (parentIssue, parentField) ->
-            val oldestVersionOnParent = getOldestVersion(parentField)
+            val oldestVersionWithKnownReleaseDateOnParent = getOldestVersionWithKnownReleaseDate(parentField)
             field
                 .filter { it !in parentField }
-                .filter { it isReleasedAfter oldestVersionOnParent }
+                .filter { it isReleasedAfter oldestVersionWithKnownReleaseDateOnParent }
                 .map { it.id }
                 .map(parentIssue.setField::partially1)
         }
 
-    private fun getOldestVersion(field: List<Version>) = field
+    private fun getOldestVersionWithKnownReleaseDate(field: List<Version>) = field
+        .filter { it.releaseDate != null }
         .sortedBy { it.releaseDate }
         .getOrNull(0)
 
+    /**
+     * Returns true only if:
+     * - The other version is `null`;
+     * - The other version has a `null` release date; OR
+     * - The current version has a non-`null` release date which is after the other version's release date.
+     */
     private infix fun Version.isReleasedAfter(other: Version?) =
-        other == null || this.releaseDate.isAfter(other.releaseDate)
+        other?.releaseDate == null ||
+                this.releaseDate?.isAfter(other.releaseDate) == true
```

Comment:
```suggestion
                this.releaseDate?.isAfter(other.releaseDate)
```
please no `== true` 😂

Reply:
This is intended, as that can be `null`